server/routes/login.js

```diff
@@ -0,0 +1,30 @@
+const router = require('express').Router();
+const { getHashedPassword } = require('../../db/db.js')
+const { getFavoritedTopics } = require('../../db/db.js')
+const passwordHash = require('password-hash');
+
+//for frontend, pass in params as `{params: {loginInfo: <input>}}`
+router.get('/', (req, res) => {
+  let login = req.query;
+
+  getHashedPassword(login, (err, data)=>{
+    if (err){
+      res.end();
+    }
+    else {
+      let validated = passwordHash.verify(login.password, data.rows[0].hashedpw);
+      if (validated === true){
+        getFavoritedTopics(data.rows[0].user_id, (err, data)=>{
+          if (err){
+            res.end();
```

Comment:
I recommend that you add a status code for error on getting a favorited topic. Good to end the connection though.

Reply:
Although that's me being pendantic. How is the rest of the back end team doing status codes on error? I'd try to come up with a style guide for that.